op-e2e/actions/proofs/isthmus_fork_test.go

```diff
@@ -133,143 +154,117 @@ func TestWithdrawalsRootIsthmus(gt *testing.T) {
-		{"AfterIsthmusWithoutWithdrawalTx", false, 0, 3},
-		{"AfterIsthmusWithWithdrawalTx", true, 3, 3},
-	}
-	for _, test := range tests {
-		t.Run(test.name, func(t *testing.T) {
-			testWithdrawlsRootIsthmus(t, test.withdrawalTx, test.withdrawalTxBlock, test.totalBlocks)
-		})
 	}
-}
-
-func testWithdrawlsRootIsthmus(gt *testing.T, withdrawalTx bool, withdrawalTxBlock, totalBlocks int) {
-	t := helpers.NewDefaultTesting(gt)
-	dp := e2eutils.MakeDeployParams(t, helpers.DefaultRollupTestParams())
-	const isthmusOffset = 2
-
-	log := testlog.Logger(t, log.LvlDebug)
-
-	dp.DeployConfig.ActivateForkAtOffset(rollup.Isthmus, isthmusOffset)
-	require.NoError(t, dp.DeployConfig.Check(log), "must have valid config")
-
-	sd := e2eutils.Setup(t, dp, helpers.DefaultAlloc)
-	_, _, _, sequencer, engine, verifier, _, _ := helpers.SetupReorgTestActors(t, dp, sd, log)
-
-	// start op-nodes
-	sequencer.ActL2PipelineFull(t)
-	verifier.ActL2PipelineFull(t)
-
-	verifyPreIsthmusHeaderWithdrawalsRoot(gt, engine.L2Chain().CurrentBlock())
-
-	ethCl := engine.EthClient()
-	for i := 1; i <= totalBlocks; i++ {
-		var tx *types.Transaction
-
-		sequencer.ActL2StartBlock(t)
 
-		if withdrawalTx && withdrawalTxBlock == i {
-			l2withdrawer, err := bindings.NewL2ToL1MessagePasser(predeploys.L2ToL1MessagePasserAddr, ethCl)
-			require.NoError(t, err, "binding withdrawer on L2")
+	isthmusOffset := 2
 
-			// Initiate Withdrawal
-			// Bind L2 Withdrawer Contract and invoke the Receive function
-			l2opts, err := bind.NewKeyedTransactorWithChainID(dp.Secrets.Alice, new(big.Int).SetUint64(dp.DeployConfig.L2ChainID))
-			require.NoError(t, err)
-			l2opts.Value = big.NewInt(500)
-			tx, err = l2withdrawer.Receive(l2opts)
-			require.NoError(t, err)
-
-			// force-include the transaction, also in upgrade blocks
-			engine.ActL2IncludeTxIgnoreForcedEmpty(dp.Addresses.Alice)(t)
+	testWithdrawlsRootIsthmus := func(gt *testing.T, testCfg *helpers.TestCfg[testCase]) {
+		t := actionsHelpers.NewDefaultTesting(gt)
+		tp := helpers.NewTestParams(func(tp *e2eutils.TestParams) {})
+		var setIsthmusTime = func(dc *genesis.DeployConfig) {
+			two := hexutil.Uint64(isthmusOffset)
+			dc.L2GenesisIsthmusTimeOffset = &two
+			dc.L1PragueTimeOffset = &zeroHex64
+		}
+		env := helpers.NewL2FaultProofEnv(t, testCfg, tp, helpers.NewBatcherCfg(), setIsthmusTime)
+		withdrawalTx, withdrawalTxBlock, totalBlocks := testCfg.Custom.withdrawalTx, testCfg.Custom.withdrawalTxBlock, testCfg.Custom.totalBlocks
+		log := testlog.Logger(t, log.LvlDebug)
+		require.NoError(t, env.Dp.DeployConfig.Check(log), "must have valid config")
+
+		sequencer, engine := env.Sequencer, env.Engine
+
+		// start op-nodes
+		sequencer.ActL2PipelineFull(t)
+
+		verifyPreIsthmusHeaderWithdrawalsRoot(gt, engine.L2Chain().CurrentBlock())
+
+		ethCl := engine.EthClient()
+		for i := 1; i <= totalBlocks; i++ {
+			var tx *types.Transaction
+
+			sequencer.ActL2StartBlock(t)
+
+			if withdrawalTx && withdrawalTxBlock == i {
```

Comment:
nit, seems we do the same boolean check twice, easier to read if we put this in a var
```suggestion
			doWithdrawalTx := withdrawalTx && withdrawalTxBlock == i
			if doWithdrawalTx {
```
and then also use `doWithdrawalTx` below.

Reply:
[8dabe04](https://github.com/ethereum-optimism/optimism/pull/15296/commits/8dabe04fc94edc1d52ca5b9ca0facf4e5e2c8751)